### PR TITLE
Dump icons in library snapshot.

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

[MAGN-9861](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9861) Dynamo Dictionary: Add path to icons, default values, search tags, and tooltips of port Library XML snapshot

Original PR: https://github.com/DynamoDS/Dynamo/pull/6406

Remove changes with icons files, since task is about Library Snapshot.

Example of xml:

```xml
<Dynamo.Search.SearchElements.ZeroTouchSearchElement>
        <FullCategoryName>Core.Color</FullCategoryName>
        <Name>ByARGB</Name>
        <Group>Create</Group>
        <Description>Construct a color by alpha, red, green, and blue components.</Description>
        <SearchTags>color</SearchTags>
        <Inputs>
          <InputParameter Name="a" Type="int" DefaultValue="255" Tooltip="The alpha value.&#xA;&#xA;int&#xA;Default value : 255" />
          <InputParameter Name="r" Type="int" DefaultValue="0" Tooltip="The red value.&#xA;&#xA;int&#xA;Default value : 0" />
          <InputParameter Name="g" Type="int" DefaultValue="0" Tooltip="The green value.&#xA;&#xA;int&#xA;Default value : 0" />
          <InputParameter Name="b" Type="int" DefaultValue="0" Tooltip="The blue value.&#xA;&#xA;int&#xA;Default value : 0" />
        </Inputs>
        <Outputs>
          <OutputParameter Name="color" Type="Color" Tooltip="Color." />
        </Outputs>
        <SmallIcon>D:\Projects\Dynamo\Dynamo_WorkingPlace\Dynamo\bin\AnyCPU\Debug\..\..\..\src\Resources\DSCoreNodes\SmallIcons\DSCore.Color.ByARGB.Small.png</SmallIcon>
        <LargeIcon>D:\Projects\Dynamo\Dynamo_WorkingPlace\Dynamo\bin\AnyCPU\Debug\..\..\..\src\Resources\DSCoreNodes\LargeIcons\DSCore.Color.ByARGB.Large.png</LargeIcon>
</Dynamo.Search.SearchElements.ZeroTouchSearchElement>
```

### Declarations

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps

### CODE REVIEW

**Comment by sharadkjaiswal** [comment]
@YanmengLi, I guess you were looking for this feature.

File: src/DynamoCore/Search/NodeSearchModel.cs [hunk 1]
```diff
@@ -48,58 +50,133 @@ internal XmlDocument ComposeXmlForLibrary()
...
+                    XmlHelper.AddAttribute(parameterNode, "Tooltip", dynamoNode.OutPorts[i].ToolTipContent);
                 }
             }
 
+            var assemblyName = Path.GetFileNameWithoutExtension(entry.Assembly);
+
+            // Get icon paths.
+            string pathToSmallIcon = Path.Combine(
+                dynamoPath,
+                @"..\..\..\src\Resources\",
```
Comment (sharadkjaiswal) [inline src/DynamoCore/Search/NodeSearchModel.cs]: This code will only work for development build and not on installed Dynamo??

**Comment by YanmengLi** [comment]
Thanks Sharad.

File: src/DynamoCore/Search/NodeSearchModel.cs (hunk 1 above)
Comment (aosyatnik) [inline src/DynamoCore/Search/NodeSearchModel.cs]: Yes. Installer doesn't provide icons files.

File: src/DynamoCore/Search/NodeSearchModel.cs [hunk 2]
```diff
@@ -48,58 +50,133 @@ internal XmlDocument ComposeXmlForLibrary()
...
-                    XmlHelper.AddAttribute(parameterNode, "Name", parameter.Item1);
-                    XmlHelper.AddAttribute(parameterNode, "Type", parameter.Item2);
+
+                    XmlHelper.AddAttribute(parameterNode, "Name", dynamoNode.InPorts[i].PortName);
+
+                    // Case for UI nodes as ColorRange, List.Create etc.
+                    // UI nodes  do not have incoming ports in NodeSearchElement, but do have incoming ports in NodeModel.
+                    // UI node initializes its incoming ports, when its constructor is called.
+                    // So, here we check if NodeModel has input ports and NodeSearchElement also has the same input ports.
+                    // UI node node will have several incoming ports on NodeModel,
```
Comment (mjkkirschner) [inline src/DynamoCore/Search/NodeSearchModel.cs]: "node node" typo?

**Comment by mjkkirschner** [comment]
@aosyatnik  - I don't want to hold this up any longer than we need to, but I would like you to think if there is another more general way to accomplish this task for Icons than putting DynamoRevit icons into the Dynamo repo....

imagine the case where a zero touch developer wanted to run this code on their own library of nodes and icons, we can of course make this improvement later.

If you think that finding a solution where the revit icons can stay is too cumbersome or time consuming than for now, lets merge this, let it get used, and then improve it. What do you think?

**Comment by aosyatnik** [comment]
@mjkkirschner , I'm fine with it. We can try to find better solution in future.

**Pull request merged**